synapse/config/emailconfig.py

```diff
@@ -74,14 +74,75 @@ def read_config(self, config):
             "account_validity", {},
         ).get("renew_at")
 
-        if self.email_enable_notifs or account_validity_renewal_enabled:
+        self.email_enable_password_reset_from_is = email_config.get(
+            "enable_password_reset_from_is", False,
+        )
+        self.enable_password_resets = (
+            self.email_enable_password_reset_from_is
+            or (not self.email_enable_password_reset_from_is and email_config != {})
+        )
+        if email_config == {} and not self.email_enable_password_reset_from_is:
+            logger.warn(
+                "User password resets have been disabled due to lack of email config."
+            )
+
+        self.email_validation_token_lifetime = email_config.get(
+            "validation_token_lifetime", 15 * 60,
+        )
+
+        if (
+            self.email_enable_notifs
+            or account_validity_renewal_enabled
+            or (self.enable_password_resets
+                and self.email_enable_password_reset_from_is)
+        ):
             # make sure we can import the required deps
             import jinja2
             import bleach
             # prevent unused warnings
             jinja2
             bleach
 
+        if self.enable_password_resets and not self.email_enable_password_reset_from_is:
+            required = [
+                "smtp_host",
+                "smtp_port",
+                "notif_from",
```

Comment:
We should probably call this something else....

Reply:
Yes, but, backwards compatibility.